Corgibytes.Freshli.Cli.Test/CommandOptions/AnalyzeCommandOptionsTest.cs

```diff
@@ -0,0 +1,104 @@
+    private const bool ExpectedCommitHistory = false;
+
+    public AnalyzeCommandOptionsTest(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    public static IEnumerable<object[]> AnalyzeOptionsArgs =>
+        new List<object[]>
+        {
+            // If passing no arguments, the default git path should be 'git'
+            new object[]
+            {
+                new[] { "analyze" }, "git", ExpectedBranch, ExpectedCommitHistory, ExpectedHistoryInterval,
+                ExpectedWorkerCount
+            },
+            // Specific git path expected
+            new object[]
+            {
+                new[] { "analyze", "--git-path", "/usr/bin/local/git" }, "/usr/bin/local/git", ExpectedBranch,
+                ExpectedCommitHistory, ExpectedHistoryInterval, ExpectedWorkerCount
+            },
+            // Specific branch expected
+            new object[]
+            {
+                new[] { "analyze", "--branch", "feature-fix-final.2.0" }, ExpectedGitPath, "feature-fix-final.2.0",
+                ExpectedCommitHistory, ExpectedHistoryInterval, ExpectedWorkerCount
+            },
+            // Entire commit history expected
+            new object[]
+            {
+                new[] { "analyze", "--commit-history" }, ExpectedGitPath, ExpectedBranch, true,
+                ExpectedHistoryInterval, ExpectedWorkerCount
+            },
+            // Three yearly history interval expected
+            new object[]
+            {
+                new[] { "analyze", "--history-interval", "3y" }, ExpectedGitPath, ExpectedBranch,
+                ExpectedCommitHistory, "3y", ExpectedWorkerCount
+            },
+            // 24 workers expected
+            new object[]
+            {
+                new[] { "analyze", "--workers", "24" }, ExpectedGitPath, ExpectedBranch, ExpectedCommitHistory,
+                ExpectedHistoryInterval, 24
+            }
+        };
+
+    [Theory]
+    [MemberData(nameof(AnalyzeOptionsArgs))]
+    public void Send_Args_ReturnsAnalyzeOptions(
+        string[] args, string expectedGitPath, string expectedBranch, bool expectedCommitHistory,
+        string expectedHistoryInterval, int expectedWorkers
+    )
+    {
+        var cmBuilder = Program.CreateCommandLineBuilder();
+        var parser = new Parser(cmBuilder.Command);
+
+        var result = parser.Parse(args);
+
+        var gitPath = result.GetOptionValueByName<string>("git-path");
+        gitPath?.Should().NotBeEmpty().And.Be(expectedGitPath);
+
+        var branch = result.GetOptionValueByName<string>("branch");
+        branch?.Should().NotBeEmpty().And.Be(expectedBranch);
+
+        var commitHistory = result.GetOptionValueByName<bool>("commit-history");
+        commitHistory.Should().Be(expectedCommitHistory);
+
+        var historyInterval = result.GetOptionValueByName<string>("history-interval");
+        historyInterval?.Should().NotBeEmpty().And.Be(expectedHistoryInterval);
```

Comment:
Same as `gitPath` and `branch`. A `null` value for `historyInterval` means that it doesn't get asserted.

Reply:
Addressed in https://github.com/corgibytes/freshli-cli/pull/212/commits/cb9a824809fa09908059b1779e63e790c6262be0